common/internal-types/src/protocol.rs

```diff
@@ -21,112 +20,114 @@ pub const DEFAULT_MAXIMUM_INCOMING_TICKET_WIN_PROB: f64 = 1.0; // TODO: change t
 /// Alias for the [`Pseudonym`](`hopr_crypto_types::types::Pseudonym`) used in the HOPR protocol.
 pub type HoprPseudonym = SimplePseudonym;
 
-/// Represents packet acknowledgement
-#[derive(Copy, Clone, Debug, PartialEq)]
+/// Unverified packet acknowledgement.
+///
+/// This acknowledgement can be serialized and deserialized to be sent over the wire.
+///
+/// To retrieve useful data, it has to be [verified](`Acknowledgement::verify`) using the public
```

Comment:
I still think there should be 2 types with TryFrom, i.e. UnverifiedAcknowledgement and Acknowledgment, where the first one is de, the latter is ser and you can TryFrom unverified to the verified.